c/self-exe.c

```diff
@@ -0,0 +1,234 @@
+/*
+  This file is meant to be standalone and suitable for use in
+  programs other than Chez Scheme.
+
+  SYNOPSIS
+    `char *S_get_process_executable_path(const char *exec_file)`
+
+  DESCRIPTION
+    `S_get_process_executable_path()` takes `exec_file` (usually
+    `argv[0]` supplied to `main()`) and returns the resolved path of
+    the containing executable. Searching `exec_file` via `PATH` is
+    used as fallback when no platform dependent method is available.
+    Memory for the result is obtained with `malloc()`, and can be
+    freed with `free()`.
+
+  RETURN VALUE
+    On success, `S_get_process_executable_path()` returns a pointer
+    to the resolved path string. Otherwise, it returns `NULL`.
+
+  NOTES
+    If `SELF_EXE_MAIN` is defined, a `main()` is defined to call and
+    print the result from `S_get_process_executable_path`, which is
+    useful for testing.
+
+  Parts of the implementaiton here are from the LLVM Project under
```

Comment:
```suggestion
  Parts of the implementation here are from the LLVM Project under
```

Reply:
Fixed while adding the license header - thanks!